pyomo/contrib/solver/base.py

```diff
@@ -0,0 +1,554 @@
+    ) -> Mapping[_GeneralVarData, float]:
+        """
+        Get mapping of variables to primals.
+
+        Parameters
+        ----------
+        vars_to_load : Optional[Sequence[_GeneralVarData]], optional
+            Which vars to be populated into the map. The default is None.
+
+        Returns
+        -------
+        Mapping[_GeneralVarData, float]
+            A map of variables to primals.
+        """
+        raise NotImplementedError(
+            f'{type(self)} does not support the get_primals method'
+        )
+
+    def _get_duals(
+        self, cons_to_load: Optional[Sequence[_GeneralConstraintData]] = None
+    ) -> Dict[_GeneralConstraintData, float]:
+        """
+        Declare sign convention in docstring here.
+
+        Parameters
+        ----------
+        cons_to_load: list
+            A list of the constraints whose duals should be loaded. If cons_to_load is None, then the duals for all
+            constraints will be loaded.
+
+        Returns
+        -------
+        duals: dict
+            Maps constraints to dual values
+        """
+        raise NotImplementedError(f'{type(self)} does not support the get_duals method')
+
+    def _get_reduced_costs(
+        self, vars_to_load: Optional[Sequence[_GeneralVarData]] = None
+    ) -> Mapping[_GeneralVarData, float]:
+        """
+        Parameters
+        ----------
+        vars_to_load: list
+            A list of the variables whose reduced cost should be loaded. If vars_to_load is None, then all reduced costs
+            will be loaded.
+
+        Returns
+        -------
+        reduced_costs: ComponentMap
+            Maps variable to reduced cost
+        """
+        raise NotImplementedError(
+            f'{type(self)} does not support the get_reduced_costs method'
+        )
+
+    @abc.abstractmethod
+    def set_instance(self, model):
+        """
+        Set an instance of the model
+        """
+
+    @abc.abstractmethod
+    def set_objective(self, obj: _GeneralObjectiveData):
+        """
+        Set current objective for the model
+        """
+
+    @abc.abstractmethod
+    def add_variables(self, variables: List[_GeneralVarData]):
+        """
+        Add variables to the model
+        """
+
+    @abc.abstractmethod
+    def add_parameters(self, params: List[_ParamData]):
+        """
+        Add parameters to the model
+        """
+
+    @abc.abstractmethod
+    def add_constraints(self, cons: List[_GeneralConstraintData]):
+        """
+        Add constraints to the model
+        """
+
+    @abc.abstractmethod
+    def add_block(self, block: _BlockData):
+        """
+        Add a block to the model
+        """
+
+    @abc.abstractmethod
+    def remove_variables(self, variables: List[_GeneralVarData]):
+        """
+        Remove variables from the model
+        """
+
+    @abc.abstractmethod
+    def remove_params(self, params: List[_ParamData]):
```

Comment:
I think this should be renamed to `remove_parameters` to be consistent with all the other methods.